### ISSUE TITLE
debugger; statements not honored in _mocha context

### ISSUE THREAD

**Issue by cdaringe** [node-inspector/node-inspector#613]
Hi all:

## problem statement
`debugger;` statements are ignored when running `node-debug _mocha some-mocha-script`.

This traditionally has not been an issue for me.  Following is my env, my script, and my results.  If there are logs/etc you would like me to provide, I would be eager to provide them per request.  Very grateful for this project.  Beats the classical debugger by a longshot :smile: 

## env
```bash
$ node-inspector -v
Node Inspector v0.9.2
$ node -v
v0.12.0
$ sourceme
CHA-CHING!  Ceez is runnin' OSX Darwin 13.4.0 64-bit (64)
```

### script
```js
'use strict';
var mocha = require('mocha');
var should = require ('should');

debugger;
console.log('bananas');
```

### results
I was going to attach screenshots, but it's frivolous.  Running ` node-debug _mocha some-mocha-script` plays right thru the script, with no stop on `debugger;`.  Running `node some-mocha-script` stops on the `debugger;` line successfully.

## supporting context - experiment 2
Manually boot this little guy
### script
```js
// debuggins.js
'use strict';
var mocha = require('mocha');
var should = require ('should');

describe('what?', function () {
    it('should do what...', function () {
        debugger;
        console.log('bananas');
    });
});
```

* `node-inspector`  process brought up and left running.  open a new terminal and try the following:
* running  `mocha --debug debuggins.js`, the script stops at the debugger *sometimes*.  but there's some race condition perhaps going on, as otherwise it stops periodically here in `timers.js`, just one line above my reported issue in #611:
![screen shot 2015-04-16 at 8 54 41 am](https://cloud.githubusercontent.com/assets/1003261/7185213/43c1c5c2-e416-11e4-8a2e-9dbfcb5456ac.png)

* running `mocha --debug-brk ./debuggins.js` produces #611 **sometimes** when it doesn't fail immediately
```bash
$ mocha --debug-brk ./debuggings.js
Debugger listening on port 5858
node.js:444
        stream = new tty.WriteStream(fd);
                 ^
TypeError: undefined is not a function
    at createWritableStdioStream (node.js:444:18)
    at process.stdout (node.js:502:16)
```

I'm not sure which of the above `mocha --debug-xxxx file` syntaxes are right (if either).

**Comment by 3y3** [comment]
Hello, @cdaringe , thanks for verbose description.
At first glance your first experiment is related with #597
The second looks like a node problem. Can you reproduce it on 0.10.*?
Or maybe using `node debug` command on 0.12.*?

I inspect the same problem at current time - `TypeError: undefined is not a function` in place where it is impossible, but I can't prepare a small bug sample to retarget this problem to nodejs.

**Comment by azachar** [comment]
Hello, I have a similar issue, and I am running on the following versions

node -v
v0.12.2

node-inspector -v
Node Inspector v0.10.0


And my command is 

```
node-debug mocha test.spec.js
```

 and I do not know how to instruct node-inspector to stop at my test. It is very frustrating...

Thanks for any help!

Cheers,
Andrej

**Comment by 3y3** [comment]
https://github.com/node-inspector/node-inspector/pull/637

**Comment by 3y3** [comment]
Fixed in 0.10.1

Next part doesn't fixed at current time, but this is out of scope of this issue.
```
$ mocha --debug-brk ./debuggings.js
Debugger listening on port 5858
node.js:444
        stream = new tty.WriteStream(fd);
                 ^
TypeError: undefined is not a function
    at createWritableStdioStream (node.js:444:18)
    at process.stdout (node.js:502:16)
```
